src/java.base/share/classes/java/net/SocksSocketImpl.java

```diff
@@ -337,7 +285,7 @@ public ProxySelector run() {
 
                 // Connects to the SOCKS server
                 try {
-                    privilegedConnect(server, serverPort, remainingMillis(deadlineMillis));
+                    delegate.connect(new InetSocketAddress(server, serverPort), remainingMillis(deadlineMillis));
```

Comment:
Hello Volkan, before this proposed change, the `privilegedConnect()` private method used to be a `synchronized` method where it used to do the `delegate.connect(...)` and then assign the input and output streams all as part of that one synchronization. With the proposed change, the synchronization is lost. 
I think we should retain the previous behaviour. One way to do that would be to introduce a private synchronized method like:

```java
private synchronized void doConnect(final String host, final int port, final int timeout)
        throws IOException {
    delegate.connect(new InetSocketAddress(host, port), timeout);
    cmdIn = getInputStream();
    cmdOut = getOutputStream();
}
```
and then call `doConnect(...)` from this line.

Reply:
It would certainly be safer to preserve the original behaviour, also in case where getInputStream/getOutputStream might throw. I like the idea of doConnect, even if Volkan's original proposal had the advantage of getting rid of the cmdIn/cmdOut instance variables that don't seem to be needed in any other method than connect.

Reply:
Agreed. Reverted to a synchronized method in 3b06bb14b653615d5095b2b20fbe672411f3e292. (Started a `tier3` run in parallel.)